src/traces/sunburst/plot.js

```diff
@@ -220,6 +220,10 @@ function plotOne(gd, cd, element, transitionOpts) {
             s.style('pointer-events', 'all');
         });
 
+        var extraAngle = trace.rotation * Math.PI / 180;
+        pt.x0 = pt.x0 + extraAngle;
+        pt.x1 = pt.x1 + extraAngle;
```

Comment:
I'm worried about modifying `x0` and `x1` this late in the process. I'm not quite sure what the adverse consequences would be, but notice `x1` does get used above in finding `nextX1ofPrevEntry`. Also if we ever optimize this such that there are some transitions during which we don't recalculate the positions, the rotation could be added multiple times.

So let's make this change before [attaching `sliceData` to any selection](https://github.com/plotly/plotly.js/blob/b889c9424a7519f743a77caffc60747795904a33/src/traces/sunburst/plot.js#L165)? Perhaps right after:

https://github.com/plotly/plotly.js/blob/b889c9424a7519f743a77caffc60747795904a33/src/traces/sunburst/plot.js#L151

We can add something like:
```js
if(trace.rotation) {
    var extraAngle = trace.rotation * Math.PI / 180;
    sliceData.forEach(function(pt) {
        pt.x0 += extraAngle;
        pt.x1 += extraAngle;
    });
}
```

Reply:
last commit (f90737b) contains this proposition.